src/libcore/iter.rs

```diff
@@ -2423,7 +2425,11 @@ macro_rules! step_impl {
             #[allow(trivial_numeric_casts)]
             fn steps_between(start: &$t, end: &$t, by: &$t) -> Option<usize> {
                 if *start <= *end {
-                    Some(((*end - *start) / *by) as usize)
+                    // Note: We assume $t <= usize/isize here
+                    Some(
+                        ((*end as isize).wrapping_sub(*start as isize) as usize
```

Comment:
Oh so this is handling the case where you're calculating the `steps_between` something like `usize::max_value() - 10` and 0? In that case I can see where the subtraction might legitimately overflow (but it's ok)!